src/QsFmt/App/Program.fs

```
@@ -33,93 +34,103 @@ let run arguments inputs =
-
-    result.MapResult(
-        (fun (options: FormatArguments) -> options |> runFormat),
-        (fun (options: UpdateArguments) -> options |> runUpdate),
-        (fun (options: UpdateAndFormatArguments) -> options |> runUpdateAndFormat),
-        (fun (_: IEnumerable<Error>) -> 2)
-    )
+    let startTime = DateTime.Now
+    let executionTime = Diagnostics.Stopwatch.StartNew()
+    use _telemetryManagerHandle = Telemetry.initializeTelemetry args
+
+    let parseArgsResult =
+        try
+            assemblyLoadContextSetup ()
+
+            CommandLine.Parser.Default.ParseArguments<FormatArguments, UpdateArguments, UpdateAndFormatArguments> args
+            |> Ok
+        with
+        | ex -> ex |> Result.Error
+
+    let commandWithOptions =
+        try
+            match parseArgsResult with
+            | Ok parsedArgs ->
+                parsedArgs.MapResult(
+                    (fun (options: IArguments) -> options |> CommandWithOptions.fromIArguments),
+                    (fun (_: IEnumerable<Error>) -> Result.Error ExitCode.BadArguments)
+                )
+                |> Ok
+            | Result.Error ex -> ex |> Result.Error
+        with
+        | ex -> ex |> Result.Error
+
+    let runResult =
+        try
+            match commandWithOptions with
+            | Ok commandWithOptions ->
+                match commandWithOptions with
+                | Ok commandWithOptions -> runCommand commandWithOptions commandWithOptions.Input
+                | Error exitCode -> { RunResult.Default with ExitCode = exitCode }
+                |> Ok
+            | Result.Error ex -> ex |> Result.Error
+        with
+        | ex -> ex |> Result.Error
+
+    Telemetry.logExecutionCompleted commandWithOptions runResult startTime executionTime.Elapsed
+
+    match runResult with
+    | Ok runResult -> runResult.ExitCode |> int
+    | Error ex ->
+        eprintf "Unexpected Error: %s" (ex.ToString())
```

@ScottCarda-MS , this is the place that you might want to throw the exception that was previously capture if you want the exception to crash the program execution at the end (after logging it).